hlf-api-matriculation.md

```diff
@@ -0,0 +1,135 @@
+    - TransactionError :: Json (refer to: [GenericError](#genericError---example))
+        => error is returned
+          If the given matriculation data could not be parsed as json.
+
+### Update Matriculation Data
+- Pass (*transient*)
+    - MatriculationData :: Json (refer to: [MatriculationData](#matriculationData))
+- Receive
+    - "" :: String
+        => Success
+    - TransactionError :: Json (refer to: [GenericError](#genericError---example))
+        => error is returned, 
+          If the matriculationID specified in the MatriculationData is not present on the ledger.
+          If the given matriculation data could not be parsed as json.
+    - TransactionError :: Json (refer to: [DetailedError](#detailedError---example))
+        => error is returned
+          If the given parameters could not be parsed.
+            If the string could not be parsed as json, the list of invalidParams is empty.
+            If some attributes within the json are not well formatted, they are listed in invalidParams.
+
+### Get Matriculation Data
+- Send
+    - matriculationId :: String
+- Receive
+    - MatriculationData :: Json (refer to: [MatriculationData](#matriculationData))
+        => Success
+    - TransactionError :: Json (refer to: [GenericError](#genericError---example))
+        => error is returned,
+          If the matriculationID is not present on the ledger.
+          If the state of data on the ledger is not consistent with the curent model.
+            This error should only occurr if the model changes while the old ledger state remains without modification.
+
+### AddEntriesToMatriculationData
+- Send
+    - matriculationId
+    - listOf(SubjectMatriculation) :: List<SubjectMatriculation> (refer to: [SubjectMatriculation](#subjectMatriculation))
+- Receive
+    - "" :: String
+      => Success
+    - TransactionError :: Json (refer to: [GenericError](#genericError---example))
+        => error is returned,
+          If the matriculationID is not present on the ledger.
+          If the state of data on the ledger is not consistent with the curent model.
+            This error should only occurr if the model changes while the old ledger state remains without modification.
+     TransactionError :: Json (refer to: [DetailedError](#detailedError---example))
+        => error is returned
+          If the given parameters could not be parsed.
+            If the string could not be parsed as json, the list of invalidParams is empty. 
+            If some attributes within the json are not well formatted, they are listed in invalidParams.
+
+## Models
+
+### MatriculationData
+```json
+{
+  "matriculationId": "string",
+  "firstName": "string",
+  "lastName": "string",
+  "birthDate": "2020-07-21",
+  "matriculationStatus": [
+    subjectMatriculationInfo :: SubjectMatriculation
+    subjectMatriculationInfo2 :: SubjectMatriculation
```

Comment:
Not sure if we decided against such notations in a previous api meeting...

Reply:
removed with new commits - not duplicating model in document, but just linking to existing models